src/braket/default_simulator/operation_helpers.py

```diff
@@ -11,11 +11,35 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
-from functools import lru_cache
-from typing import List, Tuple
+from functools import lru_cache, singledispatch
+from typing import List, Tuple, Union
 
 import numpy as np
 
+from braket.default_simulator.operation import GateOperation, KrausOperation
+
+
+def from_braket_instruction(instruction) -> Union[GateOperation, KrausOperation]:
+    """Instantiates the concrete `GateOperation` or `KrausOperation` object from the
+    specified braket instruction.
+
+    Args:
+        instruction: instruction for a circuit specified using the `braket.ir.jacqd` format.
+    Returns:
+        Union[GateOperation, KrausOperation]: instance of the concrete GateOperation or
+        KrausOperation class corresponding to the specified instruction.
+
+    Raises:
+        ValueError: If no concrete `GateOperation` or `KrausOperation` class has been
+            registered for the instruction type.
+    """
+    return _from_braket_instruction(instruction)
+
+
+@singledispatch
+def _from_braket_instruction(instruction):
+    raise ValueError(f"Instruction {instruction} not recognized")
```

Comment:
Did not know this worked!

Reply:
?

Reply:
Moving the singledispatch method to another module